### PR TITLE
STYLE: Replace pointers initialized by `new T[N]` with fixed arrays

### PR DESCRIPTION
**Pull request by N-Dekker**
Changed the type of local variables that were declared as pointer and
initialized by `new T[N]`, to the corresponding fixed size C-style array type,
`T[N]`. Removed the corresponding `delete []` statements.

Cases found by the following regular expressions:

    new \w+\[\d\]
    new \w+\[V\w+\]
    new \w+\[\w*Dimension\]

Aims to increase code readability, and may (possibly) improve the run-time
performance.

Following C++ Core Guidelines, April 10, 2022, "Prefer scoped objects, don’t
heap-allocate unnecessarily",
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#r5-prefer-scoped-objects-dont-heap-allocate-unnecessarily

### CODE REVIEW

**Review by dzenanz — APPROVED** [review]

**Comment by N-Dekker** [comment]
FYI, I'm still preparing a similar commit for the tests, so please wait. 😃  (In cases like this, when the proposed library changes are not intended to affect the behavior of the library, I think it's a good practice to keep the library changes and the test changes in separate commits.)

**Review by hjmjohnson — APPROVED** [review]
Spectacular!  Approved once added tests changes pass dashboard regression testing.

**Review by hjmjohnson — APPROVED** [review]

**Review by thewtex — APPROVED** [review]
:1st_place_medal:

**Comment by N-Dekker** [comment]
Thanks for merging, @dzenanz !

Just wondering, did you intentionally press "Create a merge commit", rather than "Rebase and merge"?  "Rebase and merge" is still my favorite option, as "Create a merge commit" creates so much extra noise in the git log history.

**Comment by dzenanz** [comment]
Just prior to merging this, I merged "update zlib" PR, which didn't have "rebase and merge" option. Clicking "rebase and merge" here didn't work, so I had to refresh the page. After refreshing I clicked "merge" quickly, without noticing in time that the option has changed to the one which creates the merge commit.

**Pull request merged**
